File: frontend/demo/component/emailfield/email-field-basic.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/email-field';
import '@vaadin/horizontal-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('email-field-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" style="align-items: baseline">
        <!-- tag::snippet[] -->
        <vaadin-email-field
          label="Email address"
          name="email"
          value="julia.scheider@email.com"
          error-message="Enter a valid email address"
          clear-button-visible
        ></vaadin-email-field>

        <vaadin-email-field
          label="Email address"
          name="email"
          value="This is not an email"
          error-message="Enter a valid email address"
          clear-button-visible
          invalid
        ></vaadin-email-field>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/emailfield/EmailFieldBasic.java
package com.vaadin.demo.component.emailfield;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.EmailField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("email-field-basic")
public class EmailFieldBasic extends HorizontalLayout {

  public EmailFieldBasic() {
    setPadding(false);

    // tag::snippet[]
    EmailField validEmailField = new EmailField();
    validEmailField.setLabel("Email address");
    validEmailField.getElement().setAttribute("name", "email");
    validEmailField.setValue("julia.scheider@email.com");
    validEmailField.setErrorMessage("Please enter a valid email address");
    validEmailField.setClearButtonVisible(true);

    EmailField invalidEmailField = new EmailField();
    invalidEmailField.setLabel("Email address");
    invalidEmailField.getElement().setAttribute("name", "email");
    invalidEmailField.setValue("This is not an email");
    invalidEmailField.setErrorMessage("Please enter a valid email address");
    invalidEmailField.setClearButtonVisible(true);
    invalidEmailField.setInvalid(true);

    add(validEmailField, invalidEmailField);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<EmailFieldBasic> { // hidden-source-line
  } // hidden-source-line
}
